layer/filestore.go

```diff
@@ -37,6 +38,16 @@ type fileMetadataTransaction struct {
 	ws    *ioutils.AtomicWriteSet
 }
 
+type fileMetadataTxData string
+
+func (txData fileMetadataTxData) GetCacheID() (string, error) {
+	return readCacheID(path.Join(string(txData), "cache-id"))
```

Comment:
Use `filepath.Join` so that this becomes OS independent. I don't think you'll need the `path` import at all.